nodes/monitor.py

```diff
@@ -80,8 +83,14 @@ def update(self, node_map):
   if save_to_csv:
     csv_writer = CSVWriter(csv_file_name, source_list)
     node_start_time = rospy.get_rostime()
-
-  this_ip = os.environ.get("ROS_IP")
+    rospy.on_shutdown(csv_writer.close)
+  else:
+    rospy.on_shutdown(lambda: rospy.loginfo("[cpu monitor] shutting down"))
```

Comment:
Could you move this as a regular loginfo call at the bottom after the main while loop? That way, the shutting down message gets printed with or without the CSV writer.

Reply:
ofc makes totally sense! I also removed the ros hostname getting, as that's more of a local thing not really in line with the oringal idea/ros guidelines probably